examples/AWSDriverExample/src/main/java/software/amazon/ReadWriteSplittingSpringJdbcTemplateExample.java

```diff
@@ -0,0 +1,114 @@
+package software.amazon;
+
+import com.zaxxer.hikari.HikariDataSource;
+import software.amazon.jdbc.PropertyDefinition;
+import javax.sql.DataSource;
+import java.sql.SQLException;
+import java.util.Properties;
+import org.springframework.jdbc.core.JdbcTemplate;
+import software.amazon.jdbc.ds.AwsWrapperDataSource;
+import software.amazon.jdbc.plugin.readwritesplitting.ReadWriteSplittingPlugin;
+
+public class ReadWriteSplittingSpringJdbcTemplateExample {
+
+  private static final String DATABASE_URL = "database-url";
+  private static final String DATABASE_NAME = "database-name";
+  private static final String USERNAME = "username";
+  private static final String PASSWORD = "password";
+  private static final int QUERY_LOOP_NUM = 32;
+
+  public static void main(String[] args) throws SQLException {
+//     scenario1();
+//     scenario2();
+    scenario3();
+  }
+
+  private static void scenario1() {
+    // given readonly not set and using simple postrges data source implementation
+    // expect no reader instance load balancing
+    DataSource dataSource = getSimplePostgresDataSource();
```

Comment:
Please use try-with-resources.
Same for other scenarios.

Reply:
I dont think DataSource or JdbcTemplate works with try-with-resources since they dont implement the AutoCloseable interface.

Reply:
After clarification, I've added a try block on `conn.setReadOnly(true)` as this throws.